### PR TITLE
#822 gRPC layer removal: modifications to generation tool and create aux API

### PR DESCRIPTION
**Pull request by mickmis**
Third batch of modifications:

- modification to the OpenAPI generation tool (python code and templates)
  - revisit authorizer interface: the previous interface would not allow to properly implement verification of scopes as per the OpenAPI standard (see previous PR #852)
  - add HTTP method in routing: the HTTP method was not taken into account previously in the routing, due to this if endpoints has the same paths supporting several methods, the request would be routed towards the wrong handler
  - fix: improperly-generated code when a response body is empty (missing package)
  - fix: in the generated server code, after routing and before invoking the handler, the context passed to the handler was `context.Background()`, i.e. the context of the request was not properly passed to the handler
  - example generated code:
    - fix: dockerfile build
    - fix: script to use the previously-generated image
    - regen and adapt generated code in the repo
  - dummy-oauth server:
    - regen and adapt generated code in the repo
- aux API:
  - create OpenAPI definition matching the previous API
  - generate server code and adapt implementation

Do note a slight API change in the `aux` API: the `getVersion` endpoint would return the version with a response `{"version": {"as_string": "<the version>"}}, this additional `as_string` is typical of grpc interfaces and I removed it. However this might be an issue if some clients already rely on this, so let me know if I should change this back.

### CODE REVIEW

File: interfaces/openapi-to-go-server/example/api/rid/interface.gen.go [hunk 1]
```diff
@@ -7,54 +7,57 @@ import (
...
-		"AuthFromAuthorizationAuthority": []api.AuthorizationOption{
-			{RequiredScopes: []string{"dss.read.identification_service_areas"}},
+	SearchIdentificationServiceAreasSecurity = []api.AuthorizationOption{
+		{
+			"AuthFromAuthorizationAuthority": {"dss.read.identification_service_areas"},
 		},
 	}
-	GetIdentificationServiceAreaSecurity = map[string]api.SecurityScheme{
-		"AuthFromAuthorizationAuthority": []api.AuthorizationOption{
-			{RequiredScopes: []string{"dss.read.position_reporting_entities"}},
```
[the remark this answers is not in the document]
Comment (BenjaminPelletier) [reply interfaces/openapi-to-go-server/example/api/rid/interface.gen.go:17]: `position_reporting_entities` are from an old version of the RID v1 API and `identification_service_areas` is desired.  But, I'm not sure where `position_reporting_entities` came from initially (were we perhaps targeting `canonical.yaml` in the RID v1 interface subrepo reference?).

[the remark this answers is not in the document]
Comment (mickmis) [reply interfaces/openapi-to-go-server/example/api/rid/interface.gen.go:17]: Intended per se no, so I assume indeed this was generated from a previous version of the OpenAPI definition.

Do note that this is the example generated code, it is not actually used anywhere.

File: interfaces/openapi-to-go-server/generate_example.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ docker container run -it \
   	-v "$(pwd)/../astm-utm/Protocol/utm.yaml:/resources/utm.yaml" \
   	-v "$(pwd)/../rid/v1/remoteid/augmented.yaml:/resources/rid.yaml" \
 	  -v "$(pwd)/example:/resources/example" \
-	  openapi-to-go-server \
+	  interuss/openapi-to-go-server \
```
[the remark this answers is not in the document]
Comment (mickmis) [reply interfaces/openapi-to-go-server/generate_example.sh:19]: Indeed thanks, fixed.

File: interfaces/openapi-to-go-server/templates/common.go.template [hunk 3]
```diff
@@ -56,6 +56,7 @@ func WriteJSON(w http.ResponseWriter, code int, obj interface{}) {
 type Handler func (exp *regexp.Regexp, w http.ResponseWriter, r *http.Request)
 
 type Route struct {
+	Method  string
```
[the remark this answers is not in the document]
Comment (mickmis) [reply interfaces/openapi-to-go-server/templates/common.go.template:59]: Do you mean the double space? That's how gofmt would format it to align the types of the fields.

File: interfaces/openapi-to-go-server/example/api/rid/interface.gen.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (barroco) [reply interfaces/openapi-to-go-server/example/api/rid/interface.gen.go:17]: If it complies with the openapi file used to generate the example, I think we are good.

File: interfaces/openapi-to-go-server/templates/common.go.template (hunk 3 above)
[the remark this answers is not in the document]
Comment (mickmis) [reply interfaces/openapi-to-go-server/templates/common.go.template]: Ah, I know what happened: this line got indented with a tab while the other ones are with 4 spaces.
I fixed it.

What is strange is that gofmt would put tabs there, but once generated they are with a tab. Maybe gofmt is invoked after generation?

[the remark this answers is not in the document]
Comment (mickmis) [reply interfaces/openapi-to-go-server/templates/common.go.template]: That's indeed the case: https://github.com/Orbitalize/dss/blob/features/822-remove-grpc-layer-gen-tool/interfaces/openapi-to-go-server/generate.py#L159

**Review by BenjaminPelletier — APPROVED** [review]

**Pull request merged**